src/main/java/efs/task/syntax/GuessNumberGame.java

```diff
@@ -13,10 +22,84 @@ public static void main(String[] args) {
     }
 
     public GuessNumberGame(String argument) {
-        //TODO: Implement the constructor
+        try {
+            this.upperBound = Integer.parseInt(argument);
+
+            if (upperBound < 1 || upperBound > UsefulConstants.MAX_UPPER_BOUND) {
+                System.out.println(UsefulConstants.WRONG_ARGUMENT);
+                throw new IllegalArgumentException();
+//                throw new IllegalArgumentException(UsefulConstants.WRONG_ARGUMENT);
+            }
+        } catch (NumberFormatException e) {
+            System.out.println(UsefulConstants.WRONG_ARGUMENT);
+            throw new IllegalArgumentException();
+//            throw new IllegalArgumentException(UsefulConstants.WRONG_ARGUMENT);
+        }
+    }
+
+    public static double log2(int n) {
+        return Math.log(n) / Math.log(2);
+    }
+
+    public void countAttempts() {
+        leftAttempts = (int) Math.floor(log2(upperBound)) + 1;
+    }
+
+    public String getLeftAttempts() {
+
+        return "[" + "*".repeat(currentAttempt + 1) + ".".repeat(leftAttempts - 1) + "]";
+    }
+
+    public boolean readAndCheckNum() {
+        Scanner scanner = new Scanner(System.in);
+
+        try {
+            System.out.println(UsefulConstants.GIVE_ME+" liczbe");
+            attemptNum = Integer.parseInt(scanner.next());
+
+            if (attemptNum > numToGuess) {
+                System.out.println("To " + UsefulConstants.TO_MUCH);
+                return false;
+            } else if (attemptNum < numToGuess) {
+                System.out.println("To " + UsefulConstants.TO_LESS);
+                return false;
+            } else {
+
+                return true;
+            }
+        } catch (NumberFormatException e) {
```

Comment:
_clean code_ za duży blok _try_. W takim bloku powinien znajdować się tylko kod rzucający wyjątkiem.

Reply:
przy próbie wynieść blok if za blok try nie przechodzi test "check if entered text is number" w workflow. chociaż lokalnie działa normalnie
![image](https://github.com/EPAM-Online-Courses/task-2-gusamyky/assets/105687329/ff183ab7-d266-4f49-94f1-b74412e799c1)


![image](https://github.com/EPAM-Online-Courses/task-2-gusamyky/assets/105687329/bf406402-64d9-461f-9ad0-57c6b960d698)